storage/innobase/lock/lock0lock.cc

```diff
@@ -1051,24 +1099,9 @@ static void lock_wait_wsrep(trx_t *trx)
          (2) If is in XA PREPARE state, it would eventually be rolled
          back and the lock conflict would be resolved when an XA COMMIT
          or XA ROLLBACK statement is executed in some other connection.
-
-         If victim has also BF status, but has earlier seqno, we have to wait.
       */
-      if (lock->trx != trx && lock->trx->mysql_thd &&
```

Comment:
The `lock->trx != trx` clause has disappeared. Was it intentional?

Reply:
Yes, the check became excessive. If the victim is not BF, then it's a different trx. Otherwise the check for trx equal to itself is in lock_has_to_wait().